coord/src/rmr/redise.c

```diff
@@ -5,38 +5,191 @@
  */
 
 #include "redise.h"
-#include "redise_parser/parse.h"
+
+typedef struct {
+  int startSlot;
+  int endSlot;
+  MRClusterNode node;
+} RLShard;
+
+static void MRTopology_AddRLShard(MRClusterTopology *t, RLShard *sh) {
+
+  int found = -1;
+  for (int i = 0; i < t->numShards; i++) {
+    if (sh->startSlot == t->shards[i].startSlot && sh->endSlot == t->shards[i].endSlot) {
```

Comment:
nit: consider adding API so we won't break `MRClusterTopology` abstraction here.